cosmrs/src/bank.rs

```diff
@@ -60,3 +60,142 @@ impl From<&MsgSend> for proto::cosmos::bank::v1beta1::MsgSend {
         }
     }
 }
+
+/// MsgMultiSend represents an arbitrary multi-in, multi-out send message.
+#[derive(Clone, Debug, Eq, PartialEq, PartialOrd, Ord)]
+pub struct MsgMultiSend {
+    /// Sender account/amount pairs.
+    pub inputs: Vec<MultiSendIO>,
+
+    /// Recipient account/amount pairs.
+    pub outputs: Vec<MultiSendIO>,
+}
+
+impl Msg for MsgMultiSend {
+    type Proto = proto::cosmos::bank::v1beta1::MsgMultiSend;
+}
+
+impl TryFrom<proto::cosmos::bank::v1beta1::MsgMultiSend> for MsgMultiSend {
+    type Error = ErrorReport;
+
+    fn try_from(proto: proto::cosmos::bank::v1beta1::MsgMultiSend) -> Result<MsgMultiSend> {
+        MsgMultiSend::try_from(&proto)
+    }
+}
+
+impl TryFrom<&proto::cosmos::bank::v1beta1::MsgMultiSend> for MsgMultiSend {
+    type Error = ErrorReport;
+
+    fn try_from(proto: &proto::cosmos::bank::v1beta1::MsgMultiSend) -> Result<MsgMultiSend> {
+        Ok(MsgMultiSend {
+            inputs: proto
+                .inputs
+                .iter()
+                .map(TryFrom::try_from)
+                .collect::<Result<_, _>>()?,
+            outputs: proto
+                .outputs
+                .iter()
+                .map(TryFrom::try_from)
+                .collect::<Result<_, _>>()?,
+        })
+    }
+}
+
+impl From<MsgMultiSend> for proto::cosmos::bank::v1beta1::MsgMultiSend {
+    fn from(coin: MsgMultiSend) -> proto::cosmos::bank::v1beta1::MsgMultiSend {
+        proto::cosmos::bank::v1beta1::MsgMultiSend::from(&coin)
+    }
+}
+
+impl From<&MsgMultiSend> for proto::cosmos::bank::v1beta1::MsgMultiSend {
+    fn from(msg: &MsgMultiSend) -> proto::cosmos::bank::v1beta1::MsgMultiSend {
+        proto::cosmos::bank::v1beta1::MsgMultiSend {
+            inputs: msg.inputs.iter().map(Into::into).collect(),
+            outputs: msg.outputs.iter().map(Into::into).collect(),
+        }
+    }
+}
+
+/// Represents a MultiSend Input or Output
+#[derive(Clone, Debug, Eq, PartialEq, PartialOrd, Ord)]
+pub struct MultiSendIO {
```

Comment:
Per Rust naming conventions, I think this should be `MultiSendIo`:

```suggestion
pub struct MultiSendIo {
```

https://rust-lang.github.io/api-guidelines/naming.html

> In UpperCamelCase, acronyms and contractions of compound words count as one word: use Uuid rather than UUID, Usize rather than USize or Stdin rather than StdIn. In snake_case, acronyms and contractions are lower-cased: is_xid_start.

See also types like `IoSlice` in `std` for precedent.